File: leaf-system/src/main/java/com/leaf/domain/LeafDictItem.java
package com.leaf.domain;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.IdView;
import org.babyfish.jimmer.sql.ManyToOne;
import org.jetbrains.annotations.Nullable;

/**
 * 字典项，与 字典表 是多对一的关系
 *
 * @author jhlz
 * @version x.x.x
 */
@Entity
public interface LeafDictItem extends BaseEntity, ID {

    /**
     * 字典 ID
     */
    @IdView("dict")
    Long dictId();

    /**
     * 字典键
     */
    String label();

    /**
     * 字典值
     */
    String value();

    /**
     * 状态：1正常，0禁用
     */
    int status();

    /**
     * 排序
     */
    int sortable();

    @Nullable
    @ManyToOne
    LeafDict dict();
}

File: src/main/kotlin/com/jhlz/domain/LeafDictItem.kt
package com.jhlz.domain

import org.babyfish.jimmer.sql.*

/**
 * 字典项，与 字典表 是多对一的关系
 *
 * @author jhlz
 * @version x.x.x
 */
@Entity
interface LeafDictItem : BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    /**
     * 字典表 id
     */
    @IdView("dict")
    val dictId: Long?

    /**
     * 字典键
     */
    val itemKey: String?

    /**
     * 字典值
     */
    val itemValue: String?

    /**
     * 状态：0正常，1禁用
     */
    val status: Int

    /**
     * 排序
     */
    val sortable: Int

    @ManyToOne
    val dict: LeafDict?
}
